server/src/test/java/org/elasticsearch/cluster/coordination/AtomicRegisterCoordinatorTests.java

```diff
@@ -529,13 +543,13 @@ private PersistentClusterState getClusterStateForTerm(long termGoal) {
             return null;
         }
 
-        private void writeHeartBeat(long term, HeartBeat heartBeat) {
-            HeartBeat previousHeartbeat = heartBeatsByTerm.put(term, heartBeat);
-            assert previousHeartbeat == null || heartBeat.leader().equals(previousHeartbeat.leader());
+        private void writeHeartBeat(HeartBeat newHeartBeat) {
+            assert heartBeat == null || heartBeat.term() <= newHeartBeat.term();
```

Comment:
I think this won't be guaranteed in the real implementation, because we will race between nodes checking the current term and writing the heartbeat blob. I'd be ok with removing it, but if we want to keep it could we add a comment that it's not strictly necessary?